src/organism.jl

```diff
@@ -0,0 +1,24 @@
+
+
+Base.@kwdef struct OrganismalPars{F}
+    α_org_dorsal::F = Param(0.8, bounds=(0.2, 1.0))
+    α_org_ventral::F = Param(0.8, bounds=(0.2, 1.0))
+    ϵ_org::F = Param(0.95, bounds=(0.1, 0.0))
+    F_sky::F = Param(0.4, bounds=(0.3, 0.5))
+    F_sub::F = Param(0.4, bounds=(0.3, 0.5))
+    p_eyes::F = Param(3e-4, bounds=(2e-4, 4e-4))
+    fO2_ext::F = Param(0.20, bounds=(0.10, 0.30))
+    rq::F = Param(0.8, bounds=(0.7, 0.9))
+    M1::F = Param(0.013, bounds=(0.01, 0.02))
+    M2::F = Param(0.8, bounds=(0.7, 0.9))
+    M3::F = Param(0.038, bounds=(0.02, 0.04))
+end
+
+Base.@kwdef struct OrganismalVars{T,P,F,B}
```

Comment:
can you explain what's happening here with T,P,F,B? I see you're distinguishing temperature, 'pressure' (though water potential is here expressed as energy content), fractions and behaviour. But I'm not clear on how these types would ultimately be used

Reply:
Uhm... they might not be needed

Reply:
I added this because I thought that it would help the compiler... not sure if it has any impact though... the logic behind them was: T for temperatures, P for psi, F for the fractions, and B for pant. I didn't have them in mind as something that would used later, but as something that might improve compilation time... maybe its just b****t. hahaha

Reply:
These help the compiler a _lot_

If you leave any fields with unspecified types the model might be 50x slower!!

But I will go through everything and performance optimise/clean up the types later on, so you probably don't need to worry too much.

Reply:
@mrke these are "parametric types" which let us not specify the exact number types (e.g they could be Float64 or Float32)

We could equally decide everything is always Float64 with specific units and lock it all in. But that can also get quite fiddly.

Type parameters can let the compile define multiple types of OrganismalVars to match the types of the fields.